homeassistant/components/yale_smart_alarm/coordinator.py

```diff
@@ -0,0 +1,120 @@
+"""DataUpdateCoordinator for the Yale integration."""
+from __future__ import annotations
+
+from datetime import timedelta
+
+from yalesmartalarmclient.client import AuthenticationError, YaleSmartAlarmClient
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import (
+    CONF_PASSWORD,
+    CONF_USERNAME,
+    STATE_LOCKED,
+    STATE_UNAVAILABLE,
+    STATE_UNLOCKED,
+)
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator
+
+from .const import DEFAULT_SCAN_INTERVAL, DOMAIN, LOGGER
+
+
+class YaleDataUpdateCoordinator(DataUpdateCoordinator):
+    """A Yale Data Update Coordinator."""
+
+    def __init__(self, hass: HomeAssistant, entry: ConfigEntry) -> None:
+        """Initialize the Verisure hub."""
+        self.entry = entry
+        self._hass = hass
+        self.yale = None
+
+        super().__init__(
+            hass,
+            LOGGER,
+            name=DOMAIN,
+            update_interval=timedelta(seconds=DEFAULT_SCAN_INTERVAL),
+        )
+
+    async def _async_update_data(self) -> dict:
+        """Fetch data from Yale."""
+
+        updates = await self._hass.async_add_executor_job(self.get_updates)
+
+        locks = []
+        door_windows = []
+
+        for lock in updates["cycle"]["data"]["device_status"]:
+            if lock["type"] == "device_type.door_lock":
```

Comment:
Invert this check and `continue` if true. Then we can outdent below.

Reply:
Sorry Martin. I don't understand what you mean. Why would it be better to invert it and check if it's not true?

Reply:
Because flat is better than nested.

Reply:
I wrote it slightly differently now, let me know